lib/index.ts

```diff
@@ -72,7 +72,7 @@ export class HUPActionHelper {
 		deviceType: string,
 		currentVersion: string,
 		targetVersion: string,
-	) {
+	): ActionName | 'takeover' {
```

Comment:
Why not just make 'takeover' another ActionName? As is, this likely forces clients to add this extra option as well. For example, see balena-proxy `services.action-backend.actions.config.ts`.

Reply:
That was the first thing I did, but I would have to create an additional item here https://github.com/balena-io-modules/balena-hup-action-utils/blob/8f8a770ff94b5aec26388e3017de41614c67f40f/lib/config.ts#L20 

For those actions `minSourceVersion` and `minTargetVersion` are required and those values don't really make sense for takeover (or I don't know what to use)

I guess I could omit `takeover` in the type definition here https://github.com/balena-io-modules/balena-hup-action-utils/blob/8f8a770ff94b5aec26388e3017de41614c67f40f/lib/types.ts#L34 but that also seems just as dirty?

I'm not entirely sure what is the best course of action

Reply:
@pipex I created PR #37 that adds a commit with changes to add 'takeover' as an action name. I didn't define an ActionConfig because that seemed simplest. But really we also could define one with empty string values for `minSourceVersion` and `minTargetVersion`. 

I think it's perfectly fine to use comments in the implementation to explain the use of 'takeover'. We _could_ modify the abstractions to accommodate how 'takeover' is used, but I don't see the value when we're not sure how our use of takeover will evolve.

As you say above, because of my implementation I modified the actions definition so an entry is optional. IMO, I would rather keep the interface to `getHUPActionType()` simple/unchanged. Let the implementation jump through whatever hoops are required to make it work.

I also added the expected `minTakeoverVersion` for the three device types.

Reply:
@kb2ma I think the problem with doing it like that is that making the ActionConfig optional removes some guarantees that are used by the code in  `getHUPActionType()`. 

The code below expects `minSourceVersion` and  `minTargetVersion` to always be defined. The type in config.ts also enforces it and the code won't compile if you add an action without those values.  

https://github.com/balena-io-modules/balena-hup-action-utils/blob/8f8a770ff94b5aec26388e3017de41614c67f40f/lib/index.ts#L143-L159

I think that if we want to use `takeover` as action we could set some nonsense values for those two variables (e.g v99.99.99) and add a comment about the reason for that configuration. 

After that we'll be able to return `takeover` from `getHUPActionType` maintaining the type guarantees

Reply:
Added a new commit with the proposed change. Included the new jetson xavier versions

Reply:
Good compromise. We always can add some metadata to ActionConfig in the future to differentiate takeover from the others. Let me try building and using this locally on my BoB.